### ISSUE TITLE
oxAuth crashed if sectorIdentifiers config is defined

### ISSUE THREAD

**Issue by iromli** [GluuFederation/gluu-flask#66]
Having `sectorIdentifiers` config set in `oxauth-static-conf.json` file (as seen [here](https://github.com/GluuFederation/community-edition-setup/blob/a900b39381ff1118f9fe102a9403c2ecdbc2c2f2/templates/oxauth-static-conf.json#L20)) crashes oxAuth (v2.4.1 build 22-Dec-2015).

Here's the error logs taken from Tomcat's `localhost.log`:

```
Feb 10, 2016 4:57:05 PM org.apache.catalina.core.StandardContext listenerStart
SEVERE: Exception sending context initialized event to listener instance of class org.jboss.seam.servlet.SeamListener
org.jboss.seam.InstantiationException: Could not instantiate Seam component: appInitializer
        at org.jboss.seam.Component.newInstance(Component.java:2170)
        at org.jboss.seam.contexts.Contexts.startup(Contexts.java:304)
        at org.jboss.seam.contexts.Contexts.startup(Contexts.java:278)
        at org.jboss.seam.contexts.ServletLifecycle.endInitialization(ServletLifecycle.java:143)
        at org.jboss.seam.init.Initialization.init(Initialization.java:744)
        at org.jboss.seam.servlet.SeamListener.contextInitialized(SeamListener.java:36)
        at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:5003)
        at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5517)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
        at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:901)
        at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:877)
        at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:652)
        at org.apache.catalina.startup.HostConfig.deployDescriptor(HostConfig.java:677)
        at org.apache.catalina.startup.HostConfig$DeployDescriptor.run(HostConfig.java:1942)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
        at java.util.concurrent.FutureTask.run(FutureTask.java:262)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
        at org.xdi.oxauth.model.config.ConfigurationFactory.getBaseDn(ConfigurationFactory.java:198)                                            [22/48]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.jboss.seam.util.Reflections.invoke(Reflections.java:22)
        at org.jboss.seam.intercept.RootInvocationContext.proceed(RootInvocationContext.java:32)
        at org.jboss.seam.intercept.SeamInvocationContext.proceed(SeamInvocationContext.java:56)
        at org.jboss.seam.transaction.RollbackInterceptor.aroundInvoke(RollbackInterceptor.java:28)
        at org.jboss.seam.intercept.SeamInvocationContext.proceed(SeamInvocationContext.java:68)
        at org.jboss.seam.core.MethodContextInterceptor.aroundInvoke(MethodContextInterceptor.java:44)
        at org.jboss.seam.intercept.SeamInvocationContext.proceed(SeamInvocationContext.java:68)
        at org.jboss.seam.async.AsynchronousInterceptor.aroundInvoke(AsynchronousInterceptor.java:52)
        at org.jboss.seam.intercept.SeamInvocationContext.proceed(SeamInvocationContext.java:68)
        at org.jboss.seam.intercept.RootInterceptor.invoke(RootInterceptor.java:107)
        at org.jboss.seam.intercept.JavaBeanInterceptor.interceptInvocation(JavaBeanInterceptor.java:185)
        at org.jboss.seam.intercept.JavaBeanInterceptor.invoke(JavaBeanInterceptor.java:103)
        at org.xdi.oxauth.model.config.ConfigurationFactory_$$_javassist_seam_3.getBaseDn(ConfigurationFactory_$$_javassist_seam_3.java)
        at org.xdi.oxauth.service.AppInitializer.loadAppliance(AppInitializer.java:410)
        at org.xdi.oxauth.service.AppInitializer.loadLdapIdpAuthConfigs(AppInitializer.java:376)
        at org.xdi.oxauth.service.AppInitializer.loadLdapAuthConfigs(AppInitializer.java:450)
        at org.xdi.oxauth.service.AppInitializer.createApplicationComponents(AppInitializer.java:110)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.jboss.seam.util.Reflections.invoke(Reflections.java:22)
        at org.jboss.seam.intercept.RootInvocationContext.proceed(RootInvocationContext.java:32)
        at org.jboss.seam.intercept.SeamInvocationContext.proceed(SeamInvocationContext.java:56)
        at org.jboss.seam.transaction.RollbackInterceptor.aroundInvoke(RollbackInterceptor.java:28)
        at org.jboss.seam.intercept.SeamInvocationContext.proceed(SeamInvocationContext.java:68)
        at org.jboss.seam.core.BijectionInterceptor.aroundInvoke(BijectionInterceptor.java:77)
        at org.jboss.seam.intercept.SeamInvocationContext.proceed(SeamInvocationContext.java:68)
        at org.jboss.seam.core.MethodContextInterceptor.aroundInvoke(MethodContextInterceptor.java:44)
        at org.jboss.seam.intercept.SeamInvocationContext.proceed(SeamInvocationContext.java:68)
        at org.jboss.seam.async.AsynchronousInterceptor.aroundInvoke(AsynchronousInterceptor.java:52)
        at org.jboss.seam.intercept.SeamInvocationContext.proceed(SeamInvocationContext.java:68)
        at org.jboss.seam.intercept.RootInterceptor.invoke(RootInterceptor.java:107)
        at org.jboss.seam.intercept.JavaBeanInterceptor.interceptInvocation(JavaBeanInterceptor.java:185)
        at org.jboss.seam.intercept.JavaBeanInterceptor.invoke(JavaBeanInterceptor.java:103)
        at org.xdi.oxauth.service.AppInitializer_$$_javassist_seam_1.createApplicationComponents(AppInitializer_$$_javassist_seam_1.java)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.jboss.seam.util.Reflections.invoke(Reflections.java:22)
        at org.jboss.seam.util.Reflections.invokeAndWrap(Reflections.java:144)
        at org.jboss.seam.Component.callComponentMethod(Component.java:2275)
        at org.jboss.seam.Component.callCreateMethod(Component.java:2198)
        at org.jboss.seam.Component.newInstance(Component.java:2158)
        ... 18 more

Feb 10, 2016 4:57:07 PM org.apache.catalina.core.ApplicationContext log
INFO: ContextListener: contextInitialized()
``

**Comment by iromli** [comment]
@qbert2k does oxAuth v2.4.1 has support for sectorIdentifiers?

**Comment by nynymike** [comment]
@iromli A client was able to register a sector identifier URI. The feature that Javier added was a way for an organization to manage which clients were associated with a certain sector_identifier_uri, and a way to publish the sector_identifier uris through oxauth.

**Comment by iromli** [comment]
@nynymike I see. 

> You may be using the wrong war file for 2.4.1

I'm using latest 2.4.1 war file from Gluu's maven repository.

> or you may want to wait for 2.4.2 which we're hoping will be available by next week in beta.

Yes, I think I need to wait for the upcoming 2.4.2 beta release.

**Comment by iromli** [comment]
Also, I will investigate in case something missing in Enterprise Edition setup

**Comment by iromli** [comment]
Switching to oxAuth v3.0.3-SNAPSHOT fixes the problem. I guess I need to wait v2.4.2 beta release.

**Comment by iromli** [comment]
Fixed in 0bc177c52da9f9251dde49e731f6635e4afc77b0
